pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -2334,3 +2335,22 @@ func isFallthroughFilterChain(fc *listener.FilterChain) bool {
 	}
 	return false
 }
+
+func removeListenerFilterTimeout(listeners []*xdsapi.Listener) {
+	for _, l := range listeners {
+		// Remove listener filter timeout for
+		// 	1. outbound listeners AND
+		// 	2. with HTTP inspector
+		hasHttpInspector := false
+		for _, lf := range l.ListenerFilters {
+			if hasHttpInspector = hasHttpInspector || lf.Name == wellknown.HttpInspector; hasHttpInspector {
```

Comment:
i think this is wrong

Reply:
Could you be more clear?

Reply:
`=` has lower priority than `||`. It will do `||` first and assignment later.

Reply:
Why not write as

```
if lf.Name == wellknown.HttpInspector {
    hasHttpInspector = true
    break
}

```